packages/sanity/src/core/studio/tasks/components/TasksSidebar/TasksSidebar.tsx

```diff
@@ -29,9 +36,98 @@ export function TasksStudioSidebar() {
         <motion.div variants={VARIANTS} transition={TRANSITION} initial="hidden" animate="visible">
           <SidebarRoot borderLeft height="fill" marginLeft={1}>
             <TasksSidebarHeader />
+            {isLoading ? <Spinner /> : <TasksList items={data ?? []} />}
           </SidebarRoot>
         </motion.div>
       )}
     </AnimatePresence>
   )
 }
+
+function TasksList({items}: {items: TaskDocument[]}) {
+  const currentUser = useCurrentUser()
+  const tabs = useMemo(() => {
+    return [
+      {
+        id: 'assigned',
+        label: 'Assigned',
+      },
+      {
+        id: 'created',
+        label: 'Created',
+      },
+      {
+        id: 'document',
+        label: 'This document',
+      },
+    ]
+  }, [])
+  const [id, setId] = useState('assigned')
+  const filteredList = useMemo(() => {
+    return items.filter((item) => {
+      if (!item.title) {
+        return false
+      }
+
+      if (id === 'assigned') {
+        return item.assignedTo === currentUser?.id
+      }
+      if (id === 'created') {
+        return item.authorId === currentUser?.id
+      }
+      if (id === 'document') {
+        return item.target?.document._ref
+      }
+      return false
+    })
+  }, [id, items, currentUser])
+
+  return (
+    <Box>
+      <Card padding={3} marginBottom={2} borderTop borderBottom>
+        <TabList space={2}>
+          {tabs.map((tab) => (
+            <Tab
+              key={`${tab.id}-tab`}
+              aria-controls={`${tab.id}-panel`}
+              id={`${tab.id}-tab`}
+              label={tab.label}
+              onClick={() => setId(tab.id)}
+              selected={id === tab.id}
+            />
+          ))}
+        </TabList>
+      </Card>
+      <List items={filteredList ?? []} />
+    </Box>
+  )
+}
+
+function List({items}: {items: TaskDocument[]}) {
+  const dateFormatter = useDateTimeFormat({
+    dateStyle: 'medium',
+  })
+  return (
+    <Box padding={3}>
+      <Stack space={3}>
+        {items
+          .filter((item) => Boolean(item.title))
+          .map((item) => (
+            <TasksListItem key={item._id}>
+              <Stack space={2}>
+                <Text size={1} weight="bold">
+                  {item.title}
+                </Text>
+                <Flex align="center" gap={1}>
+                  <CalendarIcon />
+                  <Text size={1} muted>
+                    {item.dueBy ? dateFormatter.format(new Date(item.dueBy)) : 'No due date'}
+                  </Text>
+                </Flex>
+              </Stack>
```

Comment:
Could we move this into the `TaskListItem` and let that component handle all the ui for it?